### PR TITLE
Chore/cleanup jsonapi

### PR DESCRIPTION
**Pull request by tsubik**

### CODE REVIEW

File: app/resources/v1/country_resource.rb [hunk 1]
```diff
@@ -13,7 +13,6 @@ class CountryResource < JSONAPI::Resource
     has_many :required_operator_documents
     has_many :required_gov_documents
     has_many :governments
-    has_many :monitors
```
Comment (santostiago) [inline app/resources/v1/country_resource.rb:16]: Aren't we using this in the apps? (isn't there any request on the observations tool or the FE that does countries?include=monitors)?

File: app/resources/v1/operator_document_country_history_resource.rb [hunk 2]
```diff
@@ -3,7 +3,6 @@
 module V1
   class OperatorDocumentCountryHistoryResource < OperatorDocumentHistoryResource
     has_one :required_operator_document_country
-    has_many :documents
```
Comment (santostiago) [inline app/resources/v1/operator_document_country_history_resource.rb:6]: Mmm... isn't this fetched in the FE?
To show the previous files uploaded?

File: app/resources/v1/operator_document_country_resource.rb [hunk 3]
```diff
@@ -3,7 +3,6 @@
 module V1
   class OperatorDocumentCountryResource < OperatorDocumentResource
     has_one :required_operator_document_country
-    has_many :documents
```
Comment (santostiago) [inline app/resources/v1/operator_document_country_resource.rb:6]: Don't we include this to display the documents? Do we make another request to the `documents` endpoint?

File: app/resources/v1/operator_document_fmu_history_resource.rb [hunk 4]
```diff
@@ -3,7 +3,6 @@
 module V1
   class OperatorDocumentFmuHistoryResource < OperatorDocumentHistoryResource
     has_one :required_operator_document_fmu
-    has_many :documents
```
Comment (santostiago) [inline app/resources/v1/operator_document_fmu_history_resource.rb:6]: And here

File: app/resources/v1/operator_document_fmu_resource.rb [hunk 5]
```diff
@@ -3,7 +3,6 @@
 module V1
   class OperatorDocumentFmuResource < OperatorDocumentResource
     has_one :required_operator_document_fmu
-    has_many :documents
```
Comment (santostiago) [inline app/resources/v1/operator_document_fmu_resource.rb:6]: Same here

File: config/routes.rb [hunk 6]
```diff
@@ -28,43 +28,43 @@
 
     scope '(:locale)', locale: /en|fr/ do
       # Resources
-      jsonapi_resources :about_page_entries, only: [:index, :show] do; end
```
Comment (santostiago) [inline config/routes.rb:31]: Mmmm.. I don't think this is a good idea (even though the way it's done it's a hack).
But if the block isn't specifically defined as empty, the `jsonapi_resources` will generate a lot of sub-routes that we don't need.

File: spec/integration/v1/categories_spec.rb [hunk 7]
```diff
@@ -4,24 +4,6 @@ module V1
   describe 'Categories', type: :request do
     it_behaves_like "jsonapi-resources", Category, {
       show: {},
-      create: {
```
Comment (santostiago) [inline spec/integration/v1/categories_spec.rb:7]: Mmmm.. why the removal of all these tests? Observations tool users can still perform CRUD on all these resources, right? Why remove the specs?

**Review by santostiago — COMMENTED** [review]
I have a couple of questions.
But in general, it is very DRY and cleaner.

**Review by santostiago — COMMENTED** [review]
I have a couple of questions.
But in general, it is very DRY and cleaner.

**Review by santostiago — COMMENTED** [review]
I have a couple of questions.
But in general, it is very DRY and cleaner.

**Review by santostiago — COMMENTED** [review]
I have a couple of questions.
But in general, it is very DRY and cleaner.

**Comment by santostiago** [comment]
(ok, GH got crazy and repeated the same stuff a few times :D )

File: app/resources/v1/operator_document_country_resource.rb (hunk 3 above)
Comment (tsubik) [reply app/resources/v1/operator_document_country_resource.rb:6]: What are `documents` how jsonapi could know if there is no model. The whole problem with configuration was those weird `do; end` blocks in `routes.rb` when I removed that, the project was not even compiling showing the errors like `documents` do not exist, `monitor` model does not exists etc. That's why I removed all those mistakes, there is no such thing as `document` as there is no corresponding model.

File: app/resources/v1/country_resource.rb (hunk 1 above)
Comment (tsubik) [reply app/resources/v1/country_resource.rb:16]: There is no model `monitor` that never worked and errors were silenced by `do; end` blocks in routes

File: config/routes.rb (hunk 6 above)
Comment (tsubik) [reply config/routes.rb:31]: No idea where this hack is coming from, only I know it's silencing wrongly configured resources errors. I'm removing it and doing things as in documentation.

File: spec/integration/v1/categories_spec.rb (hunk 7 above)
Comment (tsubik) [reply spec/integration/v1/categories_spec.rb:7]: I removed all routes that are not used. Categories are not created in obs tool.

Comment (tsubik) [reply spec/integration/v1/categories_spec.rb:7]: The code was commented out a long time ago.

Comment (tsubik) [reply spec/integration/v1/categories_spec.rb:7]: Also removed all not necessary API routes.

File: app/resources/v1/country_resource.rb (hunk 1 above)
Comment (tsubik) [reply app/resources/v1/country_resource.rb:16]: I have no idea why those empty blocks in `routes.rb` silenced many resource configuration errors, but I always wanted to remove those blocks as it looked just weird.

File: config/routes.rb (hunk 6 above)
Comment (tsubik) [reply config/routes.rb:31]: Do you remember why `do; end` were added to all routes?

Comment (tsubik) [reply config/routes.rb:31]: Ok, you are right, it's adding relationships routes and the only way to remove them is to add empty block. I will add them back as don't like that many endpoints. Thanks a lot :+1:

**Review by santostiago — APPROVED** [review]

**Pull request merged**